management/management_api/endpoints/endpoint_utils.py

```diff
@@ -191,28 +188,26 @@ def list_endpoints(namespace: str, id_token: str):
         deployments = apps_api_instance.list_namespaced_deployment(namespace)
     except ApiException as apiException:
         raise KubernetesGetException('endpoint', apiException)
-    endpoints_name_status = get_endpoints_name_status(deployments, namespace)
+    endpoints_name_status = get_endpoints_metadata(deployments, namespace)
     logger.info(endpoints_name_status)
     return endpoints_name_status
 
 
-def get_endpoints_name_status(deployments, namespace):
+def get_endpoints_metadata(deployments, namespace):
     deployments = deployments.items
-    name_status = "There are no endpoints present in {} tenant".format(namespace)
+    endpoints_metadata = list()

```

Comment:
why here is `list()` and there is `dict()` three lines below

Reply:
Here we've got list of endpoints metadata. Below there is a dict containing single endpoint matadata. So this list here is actually list of dictionaries, each representing single endpoint. In the loop below we append those dictionaries to the list which is then returned.

Reply:
I understand. I didn't see `s` in this earlier `endpoints_metadata`